File: epi_judge_python/test_framework/tri_bool.py
from enum import Enum, auto


class TriBool(Enum):
    FALSE = auto()
    TRUE = auto()
    INDETERMINATE = auto()

    def get_or_default(self, default_value):
        if self == self.FALSE:
            return False
        elif self == self.TRUE:
            return True
        return default_value

File: epi_judge_java/epi/test_framework/TriBool.java

package epi.test_framework;

public enum TriBool {
  FALSE,
  TRUE,
  INDETERMINATE;

  public boolean getOrDefault(boolean defaultValue) {
    switch (this) {
    case FALSE:
      return false;
    case TRUE:
      return true;
    case INDETERMINATE:
    default:
      return defaultValue;
    }
  }
}
